src/lock_commit/node.rs

```diff
@@ -99,21 +71,38 @@ impl Node {
     }
 
     /// Runs the node to process network messages incoming in the given receiver
-    pub async fn run(&mut self, mut network_receiver: NetworkReciver<Command>) {
-        while let Some((message, reply_sender)) = network_receiver.recv().await {
-            self.handle_msg(message, reply_sender).await;
+    pub async fn run(
+        &mut self,
+        mut network_receiver: Receiver<(Command, oneshot::Sender<()>)>,
+        mut client_receiver: Receiver<(Command, oneshot::Sender<CommandResult>)>,
+    ) -> JoinHandle<()> {
+        loop {
+            tokio::select! {
+            Some((command, reply_sender)) = client_receiver.recv() => {
```

Comment:
I would expect this one to be a `ClientCommand` and not a `Command::Client(ClientCommand)` how does the deserialization figure out to wrap it in the parent enum? cc @igamigo

Reply:
Up until this PR, the serialization and deserialization was done over the Command() enum, so they could be Command::Client(cmd) or Command::Network(cmd). With this PR however, that could be changed because the endpoints where you are receiving each of those variants are on 2 different ports, so you can make assumptions on each one.

Reply:
Up until this PR, the serialization and deserialization was done over the Command() enum, so they could be Command::Client(cmd) or Command::Network(cmd). With this PR however, that could be changed because the endpoints where you are receiving each of those variants are on 2 different ports, so you can make assumptions on each one.